src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicApi.cs

```diff
@@ -51,68 +51,96 @@ private MsQuicApi(QUIC_API_TABLE* apiTable)
+                if (version < MsQuicVersion)
+                {
+                    if (NetEventSource.Log.IsEnabled())
+                    {
+                        NetEventSource.Info(null, $"Incompatible MsQuic library version '{version}', expecting '{MsQuicVersion}'");
+                    }
+                    return;
+                }
+
+                // Assume SChanel is being used on windows and query for the actual provider from the library
+                QUIC_TLS_PROVIDER provider = OperatingSystem.IsWindows() ? QUIC_TLS_PROVIDER.SCHANNEL : QUIC_TLS_PROVIDER.OPENSSL;
+                size = sizeof(QUIC_TLS_PROVIDER);
+                apiTable->GetParam(null, QUIC_PARAM_GLOBAL_TLS_PROVIDER, &size, &provider);
+                UsesSChannelBackend = provider == QUIC_TLS_PROVIDER.SCHANNEL;
+
+                if (UsesSChannelBackend)
+                {
+                    // Implies windows platform, check TLS1.3 availability
+                    if (!IsWindowsVersionSupported())
                     {
-                        int arraySize = 4;
-                        uint* libVersion = stackalloc uint[arraySize];
-                        uint size = (uint)arraySize * sizeof(uint);
-                        if (StatusSucceeded(apiTable->GetParam(null, QUIC_PARAM_GLOBAL_LIBRARY_VERSION, &size, libVersion)))
+                        if (NetEventSource.Log.IsEnabled())
                         {
-                            var version = new Version((int)libVersion[0], (int)libVersion[1], (int)libVersion[2], (int)libVersion[3]);
-                            if (version >= MsQuicVersion)
-                            {
-                                Api = new MsQuicApi(apiTable);
-                                IsQuicSupported = true;
-                            }
-                            else
-                            {
-                                if (NetEventSource.Log.IsEnabled())
-                                {
-                                    NetEventSource.Info(null, $"Incompatible MsQuic library version '{version}', expecting '{MsQuicVersion}'");
-                                }
-                            }
+                            NetEventSource.Info(null, $"Current Windows version ({Environment.OSVersion}) is not supported by QUIC. Minimal supported version is {MinWindowsVersion}");
                         }
+
+                        return;
                     }
+
+                    Tls13ServerMayBeDisabled = IsTls13Disabled(isServer: true);
+                    Tls13ClientMayBeDisabled = IsTls13Disabled(isServer: false);
                 }
+
+                Api = new MsQuicApi(apiTable);
+                IsQuicSupported = true;
             }
             finally
             {
-                if (!IsQuicSupported)
+                if (!IsQuicSupported && NativeLibrary.TryGetExport(msQuicHandle, "MsQuicClose", out IntPtr msQuicClose))
```

Comment:
I guess we had the same issue before your change as well, we just never hit the right set of conditions for it to manifest and with your original change it happened on every Windows non-TLS1.3 machine.

Reply:
Yes, we did not hit it before because the OS version check was before we opened the MsQuic library. I had to move the OS check later once we knew which TLS backend MsQuic is using.

We could have still hit the issue when opening an older MsQuic version on sufficiently new Windows OS